src/main/java/com/gradle/develocity/bamboo/DevelocityPreJobAction.java

```diff
@@ -56,20 +61,25 @@ public void execute(@NotNull StageExecution stageExecution, @NotNull BuildContex
         String accessKey = credentials.getPassword();
         if (StringUtils.isBlank(accessKey)) {
             LOGGER.warn(
-                "Shared credentials with the name {} do not have password set. Environment variable {} will not be set",
-                sharedCredentialName, Constants.DEVELOCITY_ACCESS_KEY
+                    "Shared credentials with the name {} do not have password set. Environment variable {} will not be set",
+                    sharedCredentialName, Constants.DEVELOCITY_ACCESS_KEY
             );
             return;
         }
 
         injectors.stream()
-            .filter(i -> i.hasSupportedTasks(buildContext))
-            .map(i -> i.buildToolConfiguration(configuration))
-            .filter(BuildToolConfiguration::isEnabled)
-            .findFirst()
-            .ifPresent(__ ->
-                buildContext
-                    .getVariableContext()
-                    .addLocalVariable(Constants.ACCESS_KEY, accessKey));
+                .filter(i -> i.hasSupportedTasks(buildContext))
+                .map(i -> i.buildToolConfiguration(configuration))
+                .filter(BuildToolConfiguration::isEnabled)
+                .findFirst()
+                .flatMap(__ ->
+                        shortLivedTokenClient.get(configuration.getServer(), DevelocityAccessKey.of(accessKey))
+                                .map(DevelocityAccessKey::getRawAccessKey)
+                )
+                .ifPresent(shortLivedToken ->
+                        buildContext
+                                .getVariableContext()
+                                .addLocalVariable(Constants.ACCESS_KEY, shortLivedToken)
+                );
```

Comment:
🤔 maybe more readable
```suggestion
                .map(__ ->
                        shortLivedTokenClient.get(configuration.getServer(), DevelocityAccessKey.of(accessKey))
                )
                .ifPresent(shortLivedToken ->
                        buildContext
                                .getVariableContext()
                                .addLocalVariable(Constants.ACCESS_KEY, shortLivedToken.getRawAccessKey())
                );
```

Reply:
Updated